src/Course/Extend.hs

```diff
@@ -66,8 +66,12 @@ instance Extend Optional where
     (Optional a -> b)
     -> Optional a
     -> Optional b
-  (<<=) =
-    error "todo: Course.Extend (<<=)#instance Optional"
+  func <<= fa = case fa of
+    Full _ -> Full $ func fa
+    Empty  -> Empty
+  -- func <<= fa@(Full _) = Full $ func fa
+  -- _    <<= Empty       = Empty
+  -- REVIEW: both of these work, any reason to prefer one over the other?
```

Comment:
No reason other than aesthetics.

Reply:
Do you have a preference? If so, why? I don't have a strong sense of aesthetics yet, so I'm trying to build up my heuristics.

Reply:
I have a preference for the uncommented way, because I find the `@`-aliasing ugly.